### DISCUSSION TITLE
I need Help

### DISCUSSION THREAD

**Discussion by adamloeffen** [m4tt72/terminal#29] [Q&A] [answered]
This is a really cool project and I was wondering if you wouldn't mind explain how I might be able to add a command to the terminal somehow. I'm a beginner when it comes to Javascript but I can follow instructions really well... I hope you consider answering my question :) thanks in advance

**Comment by m4tt72** [comment, marked as answer]
Hello @loeffea886,

Let's create a `helloWorld` command, it will take no parameters, and will return a string in the terminal displaying `Hello World`.

First, you need to create a file in `src/utils/bin` called `helloWorld.ts`.

Then export a function `helloWorld` like the following:

```typescript
export const helloWorld = async (args: string[]): Promise<string> => {
  return 'Hello world';
};
```

Now we need to export all the functions from that file in `src/utils/bin/index.ts`:

```typescript
// ... The previous exports
export * from './helloWorld';
```

Done ;)

**Passing args**

If you want to pass arguments to the command, say you want your command to be `helloWorld [name]`, anything you pass after the `helloWorld` command in the terminal, will be available in `args` parameter.

Let's use it now:

```typescript
export const helloWorld = async (args: string[]): Promise<string> => {
  if (args.length > 0) {
    return `Hello ${args[0]}!`;
  } else {
    return 'Hello World!';
  }
};
```

Now the command `helloWorld` will print `Hello World` if there are no arguments passed, or `Hello [name]` if you pass an argument.

**Exanple**

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/20604769/178966537-9a84a3eb-9a46-47fb-b7be-3fa843193800.png">

**Reply by adamloeffen** [reply]
Thank you so much! :)
